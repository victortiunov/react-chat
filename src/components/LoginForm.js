import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui';
import TextField from 'material-ui/TextField';
import Typography from 'material-ui/Typography';
import { LinearProgress } from 'material-ui/Progress';
import Button from 'material-ui/Button';

const styles = theme => ({
  errorMessage: {
    color: 'red',
  },
  signUpButton: {
    marginTop: theme.spacing.unit * 2,
  },
});

class LoginForm extends React.Component {
  static propTypes = {
    classes: PropTypes.objectOf(PropTypes.string).isRequired,
    error: PropTypes.string,
    isFetching: PropTypes.bool.isRequired,
    onSubmit: PropTypes.func.isRequired,
  };

  static defaultProps = {
    error: '',
  };

  state = {
    username: {
      value: '',
      isValid: true,
    },
    password: {
      value: '',
      isValid: true,
    },
  };

  handleInputChange = (event) => {
    event.persist();
    const { name, value } = event.target;

    this.setState(prevState => ({
      [name]: {
        ...prevState[name],
        value,
      },
    }));
  };

  handleSubmit = (event) => {
    event.preventDefault();

    const { username, password } = this.state;

    this.props.onSubmit(username.value, password.value);
  };
  render() {
    const { classes, error, isFetching } = this.props;
    const { username, password } = this.state;

    return (
      <form onSubmit={this.handleSubmit}>
        <TextField
          required
          fullWidth
          label="Username"
          placeholder="Input your username..."
          type="text"
          name="username"
          margin="normal"
          autoComplete="username"
          value={username.value}
          onChange={this.handleInputChange}
          error={!username.isValid}
          disabled={isFetching}
        />
        <TextField
          required
          fullWidth
          label="Password"
          placeholder="Type your password..."
          type="password"
          name="password"
          margin="normal"
          autoComplete="current-password"
          value={password.value}
          onChange={this.handleInputChange}
          error={!password.isValid}
          disabled={isFetching}
        />
        {isFetching ? (
          <LinearProgress mode="indeterminate" />
        ) : (
          <Typography className={classes.errorMessage}>{error}</Typography>
        )}
        <Button
          fullWidth
          variant="raised"
          type="submit"
          color="primary"
          className={classes.signUpButton}
          disabled={isFetching}
        >
          Login
        </Button>
      </form>
    );
  }
}

export default withStyles(styles)(LoginForm);
