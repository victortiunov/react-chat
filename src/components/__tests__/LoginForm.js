/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from '../LoginForm';

const mockProps = {
  error: '',
  isFetching: false,
  onSubmit: jest.fn(),
};

describe('<LoginForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<LoginForm {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<LoginForm {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders progress bar', () => {
    const tree = renderer.create(<LoginForm {...mockProps} isFetching />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders error message', () => {
    const error = 'Errors happen!';
    const tree = renderer.create(<LoginForm {...mockProps} error={error} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
