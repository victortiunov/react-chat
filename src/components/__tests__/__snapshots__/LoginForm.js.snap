// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Username
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="username"
        autoFocus={undefined}
        className="MuiInput-input-108"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Input your username..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Password
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="current-password"
        autoFocus={undefined}
        className="MuiInput-input-108 MuiInput-inputType-112"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your password..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value=""
      />
    </div>
  </div>
  <p
    className="MuiTypography-root-114 MuiTypography-body1-123 LoginForm-errorMessage-82"
  >
    
  </p>
  <button
    className="MuiButtonBase-root-153 MuiButton-root-138 MuiButton-raised-143 MuiButton-raisedPrimary-145 MuiButton-fullWidth-152 LoginForm-signUpButton-83"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-139"
    >
      Login
    </span>
    <span
      className="MuiTouchRipple-root-156"
    />
  </button>
</form>
`;

exports[`<LoginForm /> renders error message 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Username
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="username"
        autoFocus={undefined}
        className="MuiInput-input-108"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Input your username..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Password
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="current-password"
        autoFocus={undefined}
        className="MuiInput-input-108 MuiInput-inputType-112"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your password..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value=""
      />
    </div>
  </div>
  <p
    className="MuiTypography-root-114 MuiTypography-body1-123 LoginForm-errorMessage-82"
  >
    Errors happen!
  </p>
  <button
    className="MuiButtonBase-root-153 MuiButton-root-138 MuiButton-raised-143 MuiButton-raisedPrimary-145 MuiButton-fullWidth-152 LoginForm-signUpButton-83"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiButton-label-139"
    >
      Login
    </span>
    <span
      className="MuiTouchRipple-root-156"
    />
  </button>
</form>
`;

exports[`<LoginForm /> renders progress bar 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Username
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-disabled-103 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="username"
        autoFocus={undefined}
        className="MuiInput-input-108 MuiInput-inputDisabled-110"
        defaultValue={undefined}
        disabled={true}
        id={undefined}
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Input your username..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-84 MuiFormControl-marginNormal-85 MuiFormControl-fullWidth-87"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <label
      className="MuiFormLabel-root-94 MuiInputLabel-root-88 MuiInputLabel-formControl-89 MuiInputLabel-animated-92"
      data-shrink={false}
      htmlFor={undefined}
    >
      Password
      <span
        className="MuiFormLabel-asterisk-98"
      >
         *
      </span>
    </label>
    <div
      className="MuiInput-root-100 MuiInput-disabled-103 MuiInput-fullWidth-107 MuiInput-formControl-101 MuiInput-underline-104"
    >
      <input
        aria-invalid={false}
        aria-required={true}
        autoComplete="current-password"
        autoFocus={undefined}
        className="MuiInput-input-108 MuiInput-inputDisabled-110 MuiInput-inputType-112"
        defaultValue={undefined}
        disabled={true}
        id={undefined}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your password..."
        readOnly={undefined}
        required={true}
        rows={undefined}
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiLinearProgress-root-163 MuiLinearProgress-colorPrimary-164"
    mode="indeterminate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar-171 MuiLinearProgress-barColorPrimary-172 MuiLinearProgress-bar1Indeterminate-174"
      style={Object {}}
    />
    <div
      className="MuiLinearProgress-bar-171 MuiLinearProgress-barColorPrimary-172 MuiLinearProgress-bar2Indeterminate-175"
      style={Object {}}
    />
  </div>
  <button
    className="MuiButtonBase-root-153 MuiButtonBase-disabled-154 MuiButton-root-138 MuiButton-raised-143 MuiButton-raisedPrimary-145 MuiButton-disabled-147 MuiButton-fullWidth-152 LoginForm-signUpButton-83"
    disabled={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="-1"
    type="submit"
  >
    <span
      className="MuiButton-label-139"
    >
      Login
    </span>
  </button>
</form>
`;
